src/IndexingMsg.chpl

```diff
@@ -1308,26 +1356,44 @@ module IndexingMsg
             when (DType.BigInt, DType.BigInt) {
                 var x = toSymEntry(gX,bigint);
                 var y = toSymEntry(gY,bigint);
-                if x.max_bits != -1 {
-                    mod(y.a, y.a, x.max_bits);
+                var max_size = 1:bigint;
+                var has_max_bits = x.max_bits != -1;
+                if has_max_bits {
+                    max_size <<= x.max_bits;
+                    max_size -= 1;
+                }
+                if has_max_bits {
+                    y.a &= max_size;
```

Comment:
this should be done in a forall loop over the array and use a `with ref` intent to create a task local version of `max_size`, you can look at [this PR](https://github.com/Bears-R-Us/arkouda/pull/2803/files) for reference 

This is because this operation gets turned into a forall loop kinda like the one below. See [promotion](https://github.com/Bears-R-Us/arkouda/blob/master/training/CHAPEL_TUTORIAL.md#promotion) in the chpl tutorial

```chapel
forall y in y.a {
    y &= max_size;
}
```
So `max_size` will need to be remote value forwarded to be accessible on all locales. The issue is remote value forwarding a bigint used to be pretty inefficient. To workaround this in the past we created a task local copy of `max_size`, which would look something like
```chapel
forall y in y.a with (var local_max_size = max_size) {
    y &= local_max_size;
}
```

At some point this was fixed (i don't remember exactly when), so this might not be a problem anymore. Maybe @bmcdonald3 or @ronawho knows offhand

Reply:
Looking at https://github.com/chapel-lang/chapel/blob/main/CHANGES.md#performance-optimizations--improvements-1, it looks like this was fixed in the 1.31 release, so I think as long as we support 1.30, we still need these.

Reply:
awesome! Thanks @bmcdonald3!!

Reply:
Updated